packages/shared/src/lib/interest/intrestStore.ts

```diff
@@ -0,0 +1,68 @@
+import { shallowCompare } from '$lib/shallowCompare';
+
+export interface Interest {
+	_subscribe: () => () => void;
+}
+
+interface Subscription<Arguments> {
+	args: Arguments;
+	counter: number;
+	interval?: ReturnType<typeof setInterval>;
+	lastCalled: number;
+}
+
+export class InterestStore<Arguments> {
+	private readonly subscriptions: Subscription<Arguments>[] = [];
+
+	constructor(private readonly frequency: number) {}
+
+	createInterest(args: Arguments, callback: () => void): Interest {
+		return {
+			_subscribe: () => {
+				let subscription = this.subscriptions.find((subscription) =>
+					shallowCompare(subscription.args, args)
+				);
+				if (!subscription) {
+					subscription = {
+						args,
+						counter: 0,
+						lastCalled: 0,
+						interval: undefined
+					} as Subscription<Arguments>;
+				}
+
+				this.subscriptions.push(subscription);
```

Comment:
When a previous subscription was found on lines 21:23 this line will push a duplicate. Maybe move this to line 32?

Reply:
Thanks, this is indeed a subtle problem